lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/codeactions/AddFeature.java

```diff
@@ -123,7 +123,14 @@ public void doCodeAction(ICodeActionRequest request, List<CodeAction> codeAction
             referenceRangeStart = server.getStart();
             referenceRangeEnd = server.getStartTagCloseOffset()+1;
         }
+        
         Range referenceRange = XMLPositionUtility.createRange(referenceRangeStart, referenceRangeEnd, document);
+        // check if null, clarify and reduce verbosity of error if featureManager not found.
+        // TODO: can check earlier to avoid createRange()'s error log above
+        if (referenceRange == null) {
+            LOGGER.warning("Could not find range for feature insertion. featureManager may not be defined.");
+            return;
```

Comment:
I don't understand this new code. We won't even come in here if there is no `featureManager` with the current implementation. Is this for the future?

Reply:
Removing this to debug further and address separately. Only happens in a strange scenario with an improperly XML-formatted server config xml.